File: programme/pangram-checker/pangram-checker.py
# Write a programme to check Pangram sentence

input_string = input('Input:  ')

input_string = str.lower(input_string)

alphabets = set({})

for i in input_string:
    if 'a' <= i <= 'z':
        alphabets.add(i)

if len(alphabets) == 26:
    print('Output: Pangram Sentence')
else:
    print('Output: Not a pangram sentence')

File: programme/pangram-checker/PangramChecker.java
// Write a programme to check Pangram sentence

import java.util.HashMap;
import java.util.Scanner;

public class PangramChecker {

	public static void main(String[] args) {
		String alphabet = "abcdefghijklmnopqrstuvwxyz";
		HashMap<Character, Boolean> letters = new HashMap<Character, Boolean>();
		boolean pangram = true;
		Scanner sc = new Scanner(System.in);

		// read input
		System.out.print("Input  : ");
		String input = sc.nextLine();

		// convert string to lower case
		input = input.toLowerCase();

		// iterate for each character in input
		for (int i = 0; i < input.length(); i++) {
			letters.put(input.charAt(i), true);
		}

		// iterate to check if every alphabet character is in the hashmap
		for (int i = 0; i < alphabet.length(); i++) {
			// if an alphabet character is missing then set pangam to false
			if (letters.get(alphabet.charAt(i)) == null) {
				pangram = false;
			}
		}

		System.out.print("Output : ");
		System.out.println(pangram ? "Pangram Sentence" : "Not a pangram sentence");

		sc.close();
	}
}
